modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/daemon/SelectablePeerDiscoveryDelay.scala

```diff
@@ -0,0 +1,71 @@
+package org.tessellation.sdk.infrastructure.snapshot.daemon
+
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import scala.concurrent.duration.FiniteDuration
+
+import org.tessellation.cli.AppEnvironment
+import org.tessellation.cli.AppEnvironment.Dev
+import org.tessellation.schema.node.NodeState.Ready
+import org.tessellation.schema.peer.Peer
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.snapshot.PeerDiscoveryDelay
+import org.tessellation.sdk.infrastructure.snapshot.PeerSelect.NoPeersToSelect
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosInt
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import retry.RetryPolicies.{constantDelay, limitRetriesByCumulativeDelay}
+import retry.retryingOnFailures
+
+object SelectablePeerDiscoveryDelay {
+
+  def make[F[_]: Async](
+    clusterStorage: ClusterStorage[F],
+    appEnvironment: AppEnvironment,
+    checkPeersAttemptDelay: FiniteDuration,
+    checkPeersMaxDelay: FiniteDuration,
+    additionalDiscoveryDelay: FiniteDuration,
+    minPeers: PosInt
```

Comment:
so, I can pass `minPeers = 10` but the function `waitForPeers` would succeed without any error if it would find 1 peer only. is that correct behavior?

Reply:
That is the correct behavior. On `Dev` environments, the delay only waits for 1 peer. On non-`Dev` environments, it waits for `minPeers` for up to 10 minutes. If `minPeers` peers are found, an additional 3 minute delay is executed. Otherwise, a 3 minute delay is executed followed by waiting for 1 peer for up to 10 minutes.